core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -254,6 +255,24 @@ public void setMessageCompression(boolean enabled) {
     // noop
   }
 
+  /**
+   * Defers flushing the messages written using {@link #sendMessage(Object)} to the
+   * transport until the write buffer is full. The amount of data buffered while corked is
+   * dependent on the transport implementation so users of this API should not make strong
+   * assumptions about how long data is buffered while corked.
+   *
+   * <p>Corking can provide throughput gains by allowing a transport to perform fewer writes to
+   * the underlying network.
+   *
+   * <p>The {@link ClientStream#flush()} is being deferred until all opened corks are closed.
```

Comment:
This is an internal detail. I don't think it should be part of the API.

Reply:
Ok, how about:  `In case of multiple {@link Cork}s being open, all of them need to be closed in order to continue flushing.`?

Reply:
changed